src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -306,7 +308,7 @@ public String toString(){
             }           
             if (type == LearnerType.OBSERVER) sw.append(":observer");
             else if (type == LearnerType.PARTICIPANT) sw.append(":participant");            
-            if (clientAddr!=null){
+            if (clientAddr!=null && !isClientAddrFromStatic){
```

Comment:
Not sure we should do this or not, others LGTM.

Reply:
seems good to keep the old semantic, there are some test cases that might check this. Though it should be ok to not to this too as toString is not on critical path of any functionality as far as I know.